packages/nyc-test-coverage/src/interceptors.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  proxySinks,
+  SignalInput,
+  WorkflowExecuteInput,
+  WorkflowInterceptorsFactory,
+  WorkflowInboundCallsInterceptor,
+  QueryInput
+} from '@temporalio/workflow';
+import { CoverageSinks } from './sinks';
+
+const { coverage } = proxySinks<CoverageSinks>();
+
+class CoverageInterceptor implements WorkflowInboundCallsInterceptor {
```

Comment:
You would probably be better off using internals interceptors, I’m not a 100% sure but I think you can simplify this.

Also note that if an exception is thrown in these methods you won’t get coverage

Reply:
Thinking about this s bit more, I think you could intercept internals concludeActivation instead of these 3 methods

Reply:
I'll do some research on internals interceptors and see what I can do.